src/main/java/frc/robot/utils/ThreadedGyro.java

```diff
@@ -0,0 +1,59 @@
+package frc.robot.utils;
+
+import com.kauailabs.navx.frc.AHRS;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class ThreadedGyro {
+    private final AHRS gyro;
+    private final AtomicBoolean shouldReset = new AtomicBoolean(false);
+    private volatile double lastGyro;
+    private volatile double angleAdjustment;
+    private boolean enabled = false;
+    private final ScheduledExecutorService executor;
+
+    public ThreadedGyro(AHRS gyro) {
+        this.gyro = gyro;
+        this.angleAdjustment = gyro.getAngleAdjustment();
+        this.executor = Executors.newScheduledThreadPool(1);
+    }
+
+    public void execute(){
+        executor.scheduleAtFixedRate(() -> {
+            gyro.setAngleAdjustment(getAngleAdjustment());
+            if (shouldReset.get()) {
+                gyro.reset();
+                shouldReset.set(false);
+            }
+            updateGyro();
+        },0,10, TimeUnit.MILLISECONDS);
```

Comment:
That should be a constant and probably larger (20-30 ms)

Reply:
Made it a constant. In terms of speed Bat-Chen mentioned making it 10 mills so we always have a fresh value for the next tic. I am not opposed to making it 20ms but problemby don't want to go higher or we might have pid issues